src/intervals.jl

```diff
@@ -0,0 +1,102 @@
+#= 
+This file contains code for calculating intervals for plotting purposes.
+These functions should, at minimum, take in some form of data input and
+return a tuple (min, max) of the limits corresponding to the function =#
+
+import Polynomials
+using Statistics: std, median
+using ImageFiltering
+
+"""
+    zscale(input::AbstractArray, 
+        nsamples=600;
+        contrast=0.25,
+        max_reject=0.5,
+        min_npixels=5,
+        k_rej=2.5,
+        max_iterations=5)
+
+Implementation of the `zscale` IRAF function for finding colorbar limits of `input`.
```

Comment:
Could you add a little more detail about what exactly the algorithm is supposed to do?  What would the usecase for this be - large heatmaps, images, etc?

Also, including some stuff from what appears to be a [description] of this function may be helpful:


The zscale algorithm is designed to display the image values near the median image value without the time consuming process of computing a full image histogram. This is particularly useful for astronomical images which generally have a very peaked histogram corresponding to the background sky in direct imaging or the continuum in a two dimensional spectrum.

A subset of the image is examined. Approximately `nsamples` pixels are sampled evenly over the image. The number of lines is a user parameter, nsample_lines. The pixels are ranked in brightness to form the function `I(i)` where i is the rank of the pixel and I is its value. Generally the midpoint of this function (the median) is very near the peak of the image histogram and there is a well defined slope about the midpoint which is related to the width of the histogram. At the ends of the I(i) function there are a few very bright and dark pixels due to objects and defects in the field. To determine the slope a linear function is fit with iterative rejection;

```math
I(i) = intercept + slope * (i - midpoint)
```
If more than half of the points are rejected then there is no well defined slope and the full range of the sample defines z1 and z2. Otherwise the endpoints of the linear function are used (provided they are within the original range of the sample):

```math
	z1 = I(midpoint) + (slope / contrast) * (1 - midpoint)
	z2 = I(midpoint) + (slope / contrast) * (npoints - midpoint)
```

As can be seen, the parameter contrast may be used to adjust the contrast produced by this algorithm.